torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -279,6 +279,11 @@ class CompiledNodeArgs {
   void collect(const caffe2::TypeMeta& t) {
     specialize_on_bytes(t.id());
   }
+  void collect(const std::shared_ptr<Node>& t) {
+    if (cond((bool)t)) {
+      collect(_compiler.node_calls.lookup(t));
+    }
+  }
```

Comment:
nit: do we want this one to call compiled_args on t or the caller? :thinking:

Reply:
We call compiled_args here:
https://github.com/pytorch/pytorch/blob/d064fc856e8399353189ba4c0945e87d56f04ee9/torch/csrc/autograd/functions/tensor.cpp#L185

I chose to put the call there to mirror where `apply_with_saved()` is called.  `compiled_args` created a bunch of graph inputs that must be consumed by `apply_with_saved`, so there is a symmetry between them. The call here is just collecting the ID of the node (in case it appears elsewhere in the graph).

Reply:
Sounds ok. Wonder what's the best place to leave a comment with these details.

Reply:
I'll add a comment.